docs/design/lexical_conventions/words.md

```diff
@@ -82,6 +83,15 @@ The following words are interpreted as keywords:
 -   `where`
 -   `while`
 
+## Identifiers
```

Comment:
As a general rule, I don't think we want a separate detailed design that just repeats what is in the overview. In this case, I think it would be reasonable to move overview content here, but another option is to make the overview itself the authority without any separate detailed design document.

Reply:
Happy either way, but I did see previous discussions about a concern that design docs were getting out of sync with proposal PRs. My concern was whether this was a lexical issue or a semantic one (i.e. we could emit a separate token, say `RESERVED_IDENTIFIER`), or whether semantic analysis later should check all identifiers. Put it here for now to at least trigger both these discussions.